### PR TITLE
IGNITE-23023 Sql. Improve result type for AVG function for NUMERIC types

### PR DESCRIPTION
**Pull request by korlov42**
https://issues.apache.org/jira/browse/IGNITE-23023

---
Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

**Review by zstan — APPROVED** [review]

File: modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItSqlOperatorsTest.java [hunk 1]
```diff
@@ -102,7 +102,7 @@ public void testArithmetic() {
     public void testAggregates() {
         assertExpression("COUNT(*)").returns(1L).check();
         assertExpression("SUM(val)").returns(1L).check();
-        assertExpression("AVG(val)").returns(1).check();
+        assertExpression("AVG(val)").returns(new BigDecimal("1.0000000000000000")).check();
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItSqlOperatorsTest.java:105]: we can use `setScale` method on DECIMAL value, like this
``BigDecimal.ONE.setScale(16, RoundingMode.UNNECESSARY)``.
For me, it looks more readable.
WDYT?

File: modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/datatypes/NumericAggregateTypeCoercionTest.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/**
+ * A set of test to verify behavior of type coercion for aggregates, when operand belongs to the NUMERIC type family.
+ *
+ * <p>Note: technically tests cases from this class doesn't involve {@link TypeCoercion}, yet this approach is convenient
+ * to highlight how exactly types are deducted for particular functions.
+ */
+public class NumericAggregateTypeCoercionTest extends BaseTypeCoercionTest {
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/datatypes/NumericAggregateTypeCoercionTest.java:48]: If you add AVG, SUM, MIN, MAX and ANY_VALUE, why had been skipped COUNT, EVERY and SOME?

File: modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/datatypes/utils/Types.java [hunk 3]
```diff
@@ -29,28 +29,48 @@ public final class Types {
...
+    public static final NativeType DECIMAL_26_16 = NativeTypes.decimalOf(26, 16);
+
     public static final NativeType DECIMAL_30_15 = NativeTypes.decimalOf(30, 15);
 
+    public static final NativeType DECIMAL_19_0 = NativeTypes.decimalOf(19, 0);
+    public static final NativeType DECIMAL_20_1 = NativeTypes.decimalOf(20, 1);
+    public static final NativeType DECIMAL_22_3 = NativeTypes.decimalOf(22, 3);
+    public static final NativeType DECIMAL_35_16 = NativeTypes.decimalOf(35, 16);
+
+    public static final NativeType DECIMAL_MAX_18 = NativeTypes.decimalOf(32767, 18);
```
[the remark this answers is not in the document]
Comment (korlov42) [reply modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/datatypes/utils/Types.java]: fixed, thanks!

File: modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItSqlOperatorsTest.java (hunk 1 above)
Comment (korlov42) [reply modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItSqlOperatorsTest.java]: replaced big literals with `setScale`

File: modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/datatypes/NumericAggregateTypeCoercionTest.java (hunk 2 above)
Comment (korlov42) [reply modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/datatypes/NumericAggregateTypeCoercionTest.java:48]: EVERY and SOME both are not numeric aggregates.

COUNT accepts everything and always returns BIGINT, so neither coercion nor any kind of computation is done during return type derivation. I may add test case for COUNT though to have comprehensive test coverage, if you insist

**Review by ygerzhedovich — APPROVED** [review]

**Pull request merged**